rplidar.go

```diff
@@ -249,86 +286,83 @@ func (d *Device) filterParams() (minAngleDiff float64, maxDistDiff float64) {
 }
 
 // Bounds returns the square meter bounds of the device.
-func (d *Device) Bounds(ctx context.Context) (r2.Point, error) {
+func (d *Device) Bounds(ctx context.Context) (r3.Vector, error) {
 	if d.bounds != nil {
 		return *d.bounds, nil
 	}
 	r, err := d.Range(ctx)
 	if err != nil {
-		return r2.Point{}, err
+		return r3.Vector{}, err
 	}
 	width := r * 2
 	height := width
-	bounds := r2.Point{width, height}
+	bounds := r3.Vector{width, height, 1}
 	d.bounds = &bounds
 	return bounds, nil
 }
 
-// Start requests that the device start up (start spinning).
-func (d *Device) Start(ctx context.Context) error {
+// Start requests that the device starts up and starts spinning.
+func (d *Device) Start() error {
 	d.mu.Lock()
 	defer d.mu.Unlock()
 	d.start()
 	return nil
 }
 
 func (d *Device) start() {
+	golog.Global.Debugf("starting motor")
+
 	d.started = true
 	d.driver.StartMotor()
 	d.driver.StartScan(false, true)
 	d.nodes = gen.New_measurementNodeHqArray(d.nodeSize)
 }
 
-// Stop request that the device stop (stop spinning).
-func (d *Device) Stop(ctx context.Context) error {
+// Stop request that the device stops spinning.
+func (d *Device) Stop() error {
 	d.mu.Lock()
 	defer d.mu.Unlock()
+
 	if d.nodes != nil {
 		defer func() {
 			gen.Delete_measurementNodeHqArray(d.nodes)
 			d.nodes = nil
 		}()
 	}
+	golog.Global.Debugf("stopping motor")
+
 	d.driver.Stop()
 	d.driver.StopMotor()
+	d.started = false
 	return nil
 }
 
-// Close just stops the device.
-func (d *Device) Close(ctx context.Context) error {
-	return d.Stop(ctx)
-}
-
 const defaultNumScans = 3
 
 // Scan performs a scan on the device and performs some filtering to clean up the data.
-func (d *Device) Scan(ctx context.Context, options lidar.ScanOptions) (lidar.Measurements, error) {
+func (d *Device) NextPointCloud(ctx context.Context) (pointcloud.PointCloud, error) {
 	d.mu.Lock()
 	defer d.mu.Unlock()
-	return d.scan(options)
+	return d.scan(ctx)
 }
 
-func (d *Device) scan(options lidar.ScanOptions) (lidar.Measurements, error) {
+func (d *Device) scan(ctx context.Context) (pointcloud.PointCloud, error) {
 	if !d.started {
 		d.start()
 		d.started = true
 	}
 	if !d.scannedOnce {
 		d.scannedOnce = true
 		// discard scans for warmup
-		//nolint
-		d.scan(lidar.ScanOptions{Count: 10})
```

Comment:
@jeremyhyde-viam Is there a reason why this was removed? Do I assume correctly, that the original intention was to skip the first 10 scans, since they're garbage?

Reply:
Got it. I changed the way this was implemented, but kept the warmup routine in the scan function for two reasons:
1. I assume Eric had a good reason to put it in there, most likely after realizing it's necessary. If we want to get rid of it, we should probably chat with him first.
2. Before being able to scan, there's a whole setup involved. It's easiest for now to keep this where it is.

Done.